datahub/core/permissions.py

```
@@ -161,14 +159,27 @@ def _check_actual_object_permission(self, request, view, obj):
         return True
 
 
-def get_model_action_for_view_action(http_method, view_action, many_to_many=False):
+def get_model_action_for_view_action(
+    http_method,
+    view_action,
+    many_to_many=False,
+    extra_view_to_action_mapping=None,
+    extra_many_to_many_action_mapping=None
+):
     """Gets the model action corresponding to a view action."""
     if http_method == 'OPTIONS':
         return 'read'
 
     if view_action is None:
         raise APIMethodNotAllowedException()
 
-    mapping = _MANY_TO_MANY_VIEW_TO_ACTION_MAPPING if many_to_many else _VIEW_TO_ACTION_MAPPING
+    if many_to_many:
+        mapping = _MANY_TO_MANY_VIEW_TO_ACTION_MAPPING
+        if isinstance(extra_many_to_many_action_mapping, dict):
```

Makes more sense to check if it's not None here. Mainly because if something with the wrong type is passed, it shouldn't be silently ignored.